### ISSUE TITLE
spy.reset() not working for toHaveBeenCalledTimes 

### ISSUE THREAD

**Issue by dgroh** [jasmine/jasmine#1396]
Jasmine: 2.5.2
Karma: 1.4.1

The method `toHaveBeenCalledTimes` does not get reseted when called `spy.reset()` 

sample:
```

  it("should get available years only once", fakeAsync(() => {
    // Arrange
    getAvailableYearsSpy = spyOn(service, "getAvailableYears").and.returnValue(Promise.resolve([2015,2016,2017]);

    fixture.detectChanges();

    expect(getAvailableYearsSpy).toHaveBeenCalledTimes(1); // True

    getAvailableYearsSpy.reset();

    // Act
    anyMethodWhichCallsGetAvailableYearsAgain();

    tick();

    // Assert
    expect(getPrognoseMeldungSpy).toHaveBeenCalledTimes(1); // False -> called twice
  }));
```

**Comment by slackersoft** [comment]
This seems to be working fine for me. You do need to call `.calls.reset()` not just `.reset()` on the spy to get it to work. The following spec passes fine for me.

```javascript
it('spies reset', function() {
  var foo = { bar: function() {} };
  var spy = spyOn(foo, 'bar').and.returnValue(42);

  foo.bar();
  expect(spy).toHaveBeenCalledTimes(1);

  spy.calls.reset();

  foo.bar();
  expect(foo.bar).toHaveBeenCalledTimes(1);

  foo.bar.calls.reset();

  foo.bar();
  expect(foo.bar).toHaveBeenCalledTimes(1);
});
```

Hope this helps. Thanks for using Jasmine!

**Comment by dgroh** [comment]
That is it, thanks a lot.

**Comment by SAkhil95** [comment]
Hey,

I am using  const routerSpy = jasmine.createSpyObj('Router', ['navigate']);

but when I do routerSpy.calls.reset() after expect(routerSpy.navigate).toHaveBeenCalledWith('xyz)..

I am getting error "TypeError: Cannot read property 'reset' of undefined"

**Comment by SAkhil95** [comment]
Hi @slackersoft ,

I have created ,

const routerSpy = jasmine.createSpyObj('Router', ['navigate']);

and in beforeEach I gave routerSpy.navigate.reset.. but its till not getting reset.

If I use routerSpy.calls.reset() , its throwing reset of undefined.

If I use routerSpy.navigate.reset() , its throwing router.navigatereset() is not a function.

Let me know if you've any solution for this.

Regards, 
Akhil
